File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
     let count = 0, ans = null;
  for (let num of nums) {
    if (count === 0) {
       ans = num;
       count++;
    } 
    else if (ans === num)  //count not zero and same ans so increase count
      count++;
    else count--;         //count not zero ans different, so decrease
 }
  
  return ans;
};

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int m = nums.size()/2;
        map<int, int>mp;
        for(int i = 0; i<nums.size();i++)
        {
            mp[nums[i]]++;
        }
        for(auto i: mp)
        {
            if(i.second>m)
                return i.first;
        }
        
        return 0;
    }
};